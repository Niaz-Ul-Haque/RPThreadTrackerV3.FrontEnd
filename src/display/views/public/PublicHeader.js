// #region imports
import React from 'react';
import PropTypes from 'prop-types';
import { Nav } from 'reactstrap';
// #endregion imports

const propTypes = {
	slug: PropTypes.string.isRequired,
	title: PropTypes.string
};
const defaultProps = {
	title: ''
};

const PublicHeader = (props) => {
	const {
		slug,
		title
	} = props;

	return (
		<header className="app-header navbar public-header">
			<a href="/" className="navbar-brand">RPTHREADTRACKER.COM</a>
			<Nav navbar>
				<h1 href={`/public/${slug}`} className="navbar-brand">{title}</h1>
			</Nav>
		</header>
	);
};

PublicHeader.propTypes = propTypes;
PublicHeader.defaultProps = defaultProps;
export default PublicHeader;
