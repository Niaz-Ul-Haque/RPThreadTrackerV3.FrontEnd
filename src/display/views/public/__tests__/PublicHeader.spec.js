// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import PublicHeader from '../PublicHeader';
// #endregion imports

const createTestProps = propOverrides => ({
	slug: 'my-slug',
	title: 'My Title',
	...propOverrides
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const element = shallow(<PublicHeader {...createTestProps()} />);
			expect(element).toMatchSnapshot();
		});
	});
});
