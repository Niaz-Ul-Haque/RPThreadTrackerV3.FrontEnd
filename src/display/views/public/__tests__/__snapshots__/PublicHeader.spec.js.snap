// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<Row
  className="public-header"
  tag="div"
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <a
      className="navbar-brand"
      href="/"
    >
      RPTHREADTRACKER.COM
    </a>
  </Col>
  <Col
    className="text-right"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <h1
      className="navbar-brand"
      href="/public/my-slug"
    >
      My Title
    </h1>
  </Col>
</Row>
`;
